Volume/Views/MainView/MagazinesList.swift

```diff
@@ -10,23 +10,50 @@ import Combine
 import SwiftUI
 
 struct MagazinesList: View {
-    @State private var cancellableQuery: AnyCancellable?
-    @State private var state: MainView.TabState<Results> = .loading
     @EnvironmentObject private var networkState: NetworkState
+    @State private var onOpenMagazineUrl: String?
+    @State private var openedUrl = false
+    @State private var sectionQueries: SectionQueries = (nil, nil)
+    @State private var sectionStates: SectionStates = (.loading, .loading)
+
 
     private func fetchContent(_ done: @escaping () -> Void = { }) {
-        // Simulate network request delay with Publication network request
-        // TODO: Replace with Magazines request
-        cancellableQuery = Network.shared.publisher(for: GetAllPublicationsQuery())
-            .map { data in data.publications.compactMap { $0 } }
+        guard sectionStates.featuredMagazines.isLoading || sectionStates.otherMagazines.isLoading else { return }
+        
+        fetchFeaturedMagazines(done)
+        fetchOtherMagazines()
+    }
+    
+    private func fetchFeaturedMagazines(_ done: @escaping () -> Void = { }) {
+        sectionQueries.featuredMagazines = Network.shared.publisher(for: GetFeaturedMagazinesQuery(limit: 7))
+            .map {
+                // not sure if force-unwrapping here is safe
+                $0.magazines!.map(\.fragments.magazineFields)
+            }
             .sink { completion in
                 networkState.handleCompletion(screen: .magazinesList, completion)
-            } receiveValue: {
+            } receiveValue: { magazineFields in
+                let featuredMagazines = [Magazine](magazineFields)
+                
+                withAnimation(.linear(duration: 0.1)) {
+                    sectionStates.featuredMagazines = .results(featuredMagazines)
+                }
                 done()
+            }
+    }
+    
+    private var currentSemester = "fa22"
```

Comment:
Can this be defined inside a function and not a class value? If it has to be a class value though it should be defined at the top of the class

Reply:
Probably better to write this in a `Constants` struct, e.g. 
```swift
private struct Constants {
   let currentSemester = "fa22"
}
```
Could also include other hard-coded values like the `0.1` animation duration, etc.